Sources/BranchSDK/BNCServerAPI.m

```diff
@@ -94,15 +94,19 @@ - (NSString *)getBaseURLForLinkingEndpoints {
 }
 
 - (NSString *)getBaseURL {
-    //Check if user has set a custom API base URL
-    if (self.customAPIURL) {
-        return self.customAPIURL;
-    }
     
     if (self.automaticallyEnableTrackingDomain) {
         self.useTrackingDomain = [self optedIntoIDFA];
     }
     
+    //Check if user has set a custom API base URL / custom API safetrack base url
+    if (self.useTrackingDomain && self.customSafeTrackAPIURL){
```

Comment:
So to confirm:
The `if` on line 98 is always true by default which means `useTrackingDomain` will always equal `optedIntoIDFA`. So then on line 103 the `if` checks if `opted into idfa` and `use custom safe track url`- so if using idfa and there is a custom safetrack url, use that. Else if not using idfa or no custom safe track url, use custom endpoint if available.

I think we'll want to simplify this logic further down.

Reply:
@gdeluna-branch Thats used for unit tests. we set this to NO in unit tests. Otherwise value set for self.useTrackingDomain is overriden by actual opted in status.